src/Tizen.Data.Tdbc/Tizen.Data.Tdbc/TransactionGuard.cs

```diff
@@ -0,0 +1,81 @@
+﻿/*
+ * Copyright (c) 2023 Samsung Electronics Co., Ltd All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the License);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+
+namespace Tizen.Data.Tdbc
+{
+    /// <summary>
+    /// This class helps the operation of the statement in a transacted way.
+    /// </summary>
+    /// <since_tizen> 11 </since_tizen>
+    public class TransactionGuard : IDisposable
+    {
+        private Tdbc.Statement _stmt;
+        private bool _commited;
+
+        /// <summary>
+        /// The boolean variable if the transact operation enabled.
```

Comment:
The flag representing whether the TransactionGuard is enabled. The TransactionGuard has no effect unless it is enabled.

Reply:
Done.